### DISCUSSION TITLE
Ordering fields in data schema

### DISCUSSION THREAD

**Discussion by vincent-dube-bistodeau** [frictionlessdata/frictionlessdata.io#740] [Q&A] [answered]
Hi guys, let me begin by saying thank you for the amazing work on this library. Truly some fine work.
I have a simple question but I can't find the answer anywhere, hope you don't mind me asking here! :)

I am using the describe function on an Excel file to infer the metadata and save it as YAML file.
Everything is fine, I can re-use the resource.yaml file, load a new set of the same data and validate it.

My only concern is that in the resource.yaml file, in the schema/fields section, 'type' comes before 'name'.
I am planning to use this file as basic documentation for my team and that hurt readability a lot.

**How can I automatically change the order of the fields?**

Thank you for your time and again great library! :)

**Comment by rgaiacs** [comment]
Hello @vincent-dube-bistodeau !

You can open the `resource.yaml` file with a text editor and re-order the fields manually. For example,

```
$ cat mwe.csv 
cat mwe.csv 
id,name
1,Jane
2,Joe
$ frictionless describe mwe.csv > resource.yaml
$ cat resource.yaml 
# --------
# metadata: mwe.csv
# --------

path: mwe.csv
name: mwe
profile: tabular-data-resource
scheme: file
format: csv
hashing: md5
encoding: utf-8
schema:
  fields:
    - type: integer
      name: id
    - type: string
      name: name
```

Now, edit `resource.yaml` with Visual Studio or another text editor.

**Comment by shashigharti** [comment, marked as answer]
Hi @vincent-dube-bistodeau 

It will be fixed in v5

https://github.com/frictionlessdata/frictionless-py/issues/1146

**Reply by vincent-dube-bistodeau** [reply]
Thank you very much! The thread is very clear! :)
Do we have an expected time release for v5?

**Reply by roll** [reply]
Hi @vincent-dube-bistodeau 
We're approaching early August for the public beta and September for the final one - https://software.frictionlessdata.io/pages/roadmap.html

**Comment by shashigharti** [comment]
It is planned around August. I will update here with confirmed dates later.
